src/WebJobs.Script.Grpc/WorkerInvocationMetrics.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc
+{
+    public class WorkerInvocationMetrics
+    {
+        public int TotalInvocations { get; set; }
+
+        public int SuccessfulInvocations { get; set; }
+
+        public double AverageInvocationLatency { get; set; }
+
+        public static int IncrementTotalInvocationsOfWorker(string workerId, IDictionary<string, WorkerInvocationMetrics> invocationMetricsPerWorkerId)
+        {
+            invocationMetricsPerWorkerId.TryGetValue(workerId, out WorkerInvocationMetrics workerInvocationMetrics);
+            workerInvocationMetrics = workerInvocationMetrics ?? new WorkerInvocationMetrics();
+
+            workerInvocationMetrics.TotalInvocations = workerInvocationMetrics.TotalInvocations + 1;
+            invocationMetricsPerWorkerId[workerId] = workerInvocationMetrics;
```

Comment:
Is this code thread safe?  I see we are using Dictionary type which is not thread safe. Should we use ConcurrentDictionary?

Reply:
Updated to use concurrentDict